core/amber/src/main/scala/edu/uci/ics/amber/engine/common/storage/DatasetFileDocument.scala

```diff
@@ -19,12 +28,13 @@ class DatasetFileDocument(fileFullPath: Path) extends VirtualDocument[Nothing] {
     )
 
   override def asInputStream(): InputStream = {
-    fileRelativePath match {
-      case Some(path) =>
-        DatasetResource.getDatasetFile(dataset.getDid, datasetVersion.getDvid, path)
-      case None =>
-        throw new IllegalArgumentException("File relative path is missing.")
-    }
+    val datasetAbsolutePath = PathUtils.getDatasetPath(UInteger.valueOf(did))
+    GitVersionControlLocalFileStorage
+      .retrieveFileContentOfVersionAsInputStream(
+        datasetAbsolutePath,
+        datasetVersionHash,
```

Comment:
datasetAbsolutePath and resolved fileAbsolutePath have redundant information (both contain the datasetAbsolutePath). it will be good to remove one of it. You can do it in future PRs.

Reply:
will do it in future PR